### PR TITLE
ACS-8351 Create agents avatar endpoint with mocked response.

### PR DESCRIPTION
**Pull request by aonych**
As per title.

### CODE REVIEW

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/config/HxInsightClientConfig.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/config/HxInsightClientConfig.java:46]: Given this was previously set differently then is it actually being used? Can we just delete this entry?

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:48]: There's currently no logging that I can see. I suggest adding something at trace or debug.

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:80]: Please can we list the checked exceptions here. I assume they're all to do with IO? We can then convert the checked exceptions into the correct exceptions so that they return the right status code (e.g. in the future we will want to return a status code from Hx Insight).

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:59]: There's not a huge amount to unit test in this PR, but I think this is something that we could add a test for.

**Review by tpage-alfresco — CHANGES_REQUESTED** [review]
Great that you've been able to get up to speed with the project so quickly. A few changes to make and then I think we can move on to interacting with the Hx Insight API.

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelationTest.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelationTest.java:62]: This unit test will fail if Wikipedia goes offline. Probably not very likely, but I don't like the test depending on an external resource like that. IMO it's fine to either delete the test for now, or extract the downloading code and use a mock in this test. I prefer the second option.

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (tpage-alfresco) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:86]: The last time we discussed this in the team we agreed upon 503 for this issue:
```
throw new WebScriptException(SC_SERVICE_UNAVAILABLE, "Failed to ask question", e);
```

I noticed that we have no consistency over it at the moment though, so I'll create a ticket to fix it in all endpoints.

Comment (tpage-alfresco) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:86]: More generally it would be better to put this code in `HxInsightClient`. Have a look at what we're doing for the other endpoints.

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (Pawel-608) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:81]: is it necessary to first write response to file and then read it?

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (krdabrowski) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:80]: Should we verify content type of received file, or we don't care and expect that HxI will always send a PNG? 🤔

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/config/HxInsightClientConfig.java
Comment (aonych) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/config/HxInsightClientConfig.java]: Deleted 👍

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (aonych) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:87]: To be honest, this is just a sample method to test if the API will work with BinaryResource and if it will return an image instead of json. I want to get a green light from you first if the API looks good to start working on the logic part. 😃

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (aonych) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:86]: Same as above 😄

Comment (krdabrowski) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:85]: Ok, in this case LGTM

**Review by krdabrowski — APPROVED** [review]

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (Pawel-608) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:86]: but we won't merge it like that🤞?

Comment (tpage-alfresco) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java:86]: I think this is fine to merge at the moment. It's going to need this code replacing and extracting to the `HxInsightClient` and we also need to add auth. Then we'll need an e2e test too.

**Review by tpage-alfresco — APPROVED** [review]
LGTM. @aonych Would you like me to create some subtasks for the ticket or are you happy you know what you need to do?

**Review by dsibilio — APPROVED** [review]

**Comment by aonych** [comment]
> LGTM. @aonych Would you like me to create some subtasks for the ticket or are you happy you know what you need to do?

@tpage-alfresco It would be great!

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelationTest.java
Comment (aonych) [reply hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelationTest.java]: Updated.

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java
Comment (aonych) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java]: Updated

**Pull request merged**
